test/spmd/test_spmd_debugging.py

```diff
@@ -116,6 +119,8 @@ def test_single_host_partial_replication_tpu(self):
     from torch_xla.distributed.spmd.debugging import visualize_tensor_sharding
     device = xm.xla_device()
     num_devices = self.n_devices
+    if num_devices != 8:
+      self.skipTest("skip num_devices!=8 env to avoid circular reasoning")
```

Comment:
Could we make these checks into test decorators? e.g. `unittest.skipIf(xr.global_runtime_device_count() != 8)`

Also for my understanding, what's the circular reasoning here?

Reply:
+1, `unittest.skipIf` is the way to go. What's the reason behind this check? `!=8` seems too restrictive -- so if we can generalize to `n` devices, it would be better?

Reply:
Thanks for both suggestion, has updated to more generalized way to generate expected table without restrictive limitation

and for `circular reasoning`, I want to describe a test situation that we need to provide test example by our function, and provide expected example by our function too,

for this `test/spmd/test_spmd_debugging.py`, if we want to test on given device kind, we need to generate table with our function, and generate expected table with our function too, so to avoid this(circular reasoning), we want to limit the test device kind to 8-devices

let me change to `code test code self` for better description